src/main.js

```diff
@@ -218,49 +218,68 @@ app.whenReady().then(async () => {
 		}
 	})
 
+	/** To store pending "trust certificate" dialogs on Linux */
+	const openMessageBoxes = new Map()
 	app.on('certificate-error', (event, webContents, url, error, certificate, callback) => {
 		event.preventDefault()
 
 		if (isLinux) {
-			return callback(process.env.NODE_ENV !== 'production')
+			const trustedFingerprints = getAppConfig('trustedFingerprints') ?? []
+			if (trustedFingerprints.includes(certificate.fingerprint)) {
+				callback(true)
+			} else {
+				let messageBox = null
+				if (openMessageBoxes.has(certificate.fingerprint)) {
+					messageBox = openMessageBoxes.get(certificate.fingerprint)
+				} else {
+					messageBox = dialog.showMessageBox(mainWindow, {
+						type: 'warning',
+						title: 'Security Warning',
+						detail:
+							[
+								`Error: ${error}`,
+								'',
+								`Subject: ${certificate.subjectName}`,
+								'',
+								`Issuer: ${certificate.issuerName ?? 'UNKNOWN'}`,
+								`- Organisations: ${certificate.issuer.organizations.join(', ')}`,
+								`- Organisation units: ${certificate.issuer.organizationUnits.join(', ')}`,
+								`- Country: ${certificate.issuer.country}`,
+								`- State: ${certificate.issuer.state}`,
+								`- Locality: ${certificate.issuer.locality}`,
+								'',
+								`Fingerprint: ${certificate.fingerprint}`,
+								'',
+								`Valid from: ${new Date(certificate.validStart * 1_000).toLocaleDateString()}`,
+								`Valid until: ${new Date(certificate.validExpiry * 1_000).toLocaleDateString()}`,
+								'',
+								'Do you trust this certificate?',
+							].join('\n'),
+						buttons: ['Yes', 'Cancel'],
+					})
+					openMessageBoxes.set(certificate.fingerprint, messageBox)
+				}
+				messageBox.then(({ response }) => {
+					if (response === 0) {
+						if (openMessageBoxes.delete(certificate.fingerprint)) { // only updates the AppConfig once
+							setAppConfig('trustedFingerprints', [certificate.fingerprint, ...trustedFingerprints])
+						}
+						callback(true)
+					} else {
+						callback(false)
+					}
```

Comment:
This works well when the certificate is accepted:
- Promise is deleted
- Config is updated
- Certificate is accepted

But when it isn't (User pressed "Cancel"), just nothing happens.
The app just doesn't work, all the following requests are blocked.
And there are no dialogs anymore, because it is still in `openMessageBoxes`.
The user must restart the app.

And maybe the user did it just by accident.

It should:
1. Either show the dialog again on the following requests
2. Or prevent the user from using the app

IMO, 1 is much simpler. 

Let's always remove the promise once it is resolved. Not only if it was resolved with a 0 response.